src/core/prometheus.cc

```diff
@@ -603,7 +604,18 @@ future<> write_text_representation(output_stream<char>& out, const config& ctx,
                     s << "\n";
                 } else {
                     add_name(s, name, value_info.id.labels(), ctx);
-                    s << to_str(value);
+                    std::string value_str;
+                    try {
+                        value_str = to_str(value);
+                    } catch (const std::range_error& e) {
+                        seastar_logger.warn("prometheus: write_text_representation: {}: {}", s.str(), e.what());
```

Comment:
This happened due to to_str erroneously calling `ui()` for `DERIVE` rather than `i()`.
I agree it's not something that users can do anything about.
It would be helpful more for debugging purposes so we can indeed print it in debug level.